.github/workflows/leetcode_directory_writer.yml

```diff
@@ -18,13 +18,22 @@ jobs:
       - name: Add python dependencies
         run: |
           pip install requests
-      - name: Write leectode DIRECTORY.md
+      - name: Write LeetCode DIRECTORY.md
         run: |
           python3 scripts/leetcode_directory_md.py 2>&1 | tee leetcode/DIRECTORY.md
-          git config --global user.name github-actions[bot]
-          git config --global user.email 'github-actions@users.noreply.github.com'
-      - name: Update LeetCode's directory
+          git pull || true
+      - name: Commit and push changes
+        uses: stefanzweifel/git-auto-commit-action@v4
+        id: commit-push
+        with:
+          commit_message: "docs: updating `leetcode/DIRECTORY.md`"
+          branch: "leetcode-directory-${{ github.sha }}"
+          create_branch: true

```

Comment:
Also I really think that we should open a PR from a separate branch when a branch gets merged...

Reply:
> I think we should stick to actions that have been thoroughly checked (audited) by us😅 I don't think committing needs a separate action...

Oh, I'm using that to create a PR if there are changes (see the check below).
And, I tested the action and it works well, with no errors or issues.

> Also I really think that we should open a PR from a separate branch when a branch gets merged...

What do you mean exactly? 🤔 
I think that's what I'm doing currently.

Reply:
Yes you're creating a separate PR but you're merging that PR directly - that means there is actually no point in creating a new PR (which is why my comment might have been confusing). I think the merge should only be done by the maintainers rather than the bot 😄

Reply:
Ahh, I get your point. I merge the PR immediately so that the LeetCode directory is updated ASAP (a PR is needed due to the branch protection). We can remove that if wanted, though.
@alexpantyukhin, what are your thoughts on this? 🤔

Reply:
I think @Panquesito7  is right. The leetcode `Directory` is better to be updated asap, so to exclude inconsistency. Also this routine is pretty straight and maintainer attention is not required there. Of course probably the situation when the leetcode API could be unreached or broken. This case need to be Highlighted.

Reply:
Consider this question: what is the difference between creating a PR and merging it immediately vs committing directly to master?

Reply:
We cannot commit directly to `master` as it is a protected branch, that's why the PR is needed.

Reply:
Hm.. this case made us to make a PR anyway.

Reply:
Will be merging for now. We can change or discuss this later on.